File: contracts/src/main/java/my/restate/e2e/services/ListObject.java
// Copyright (c) 2024 - Restate Software, Inc., Restate GmbH
//
// This file is part of the Restate e2e tests,
// which are released under the MIT license.
//
// You can find a copy of the license in file LICENSE in the root
// directory of this repository or package, or at
// https://github.com/restatedev/e2e/blob/main/LICENSE

package my.restate.e2e.services;

import dev.restate.sdk.ObjectContext;
import dev.restate.sdk.annotation.Handler;
import dev.restate.sdk.annotation.VirtualObject;
import java.util.List;

@VirtualObject(name = "ListObject")
public interface ListObject {

  @Handler
  void append(ObjectContext context, String value);

  @Handler
  List<String> get(ObjectContext context);

  @Handler
  List<String> clear(ObjectContext context);
}

File: sdk-tests/src/main/kotlin/dev/restate/sdktesting/contracts/ListObject.kt
// Copyright (c) 2023 - Restate Software, Inc., Restate GmbH
//
// This file is part of the Restate SDK Test suite tool,
// which is released under the MIT license.
//
// You can find a copy of the license in file LICENSE in the root
// directory of this repository or package, or at
// https://github.com/restatedev/sdk-test-suite/blob/main/LICENSE
package dev.restate.sdktesting.contracts

import dev.restate.sdk.annotation.*

@VirtualObject
@Name("ListObject")
interface ListObject {
  /** Append a value to the list object */
  @Handler suspend fun append(value: String)

  /** Get current list */
  @Handler suspend fun get(): List<String>

  /** Clear list */
  @Handler suspend fun clear(): List<String>
}
